Stoica Tudor/problema vega/codul fara comentarii.txt

```diff
@@ -0,0 +1,364 @@
+                }
+                printf("\n \n \n \n \n ");
+                i=start;
+            }
+        }
+
+        if(isFor(input,i)==true || isDo(input,i)==true)
+        {
+            cnt++;
+
+            while(input[i]!='{')
+            {
+                i++;
+            }
+
+            start=i;
+            stack++;
+
+            while(stack>0)
+            {
+                i++;
+
+                if(input[i]=='{')
+                {
+                    stack++;
+                }
+
+                if(input[i]=='}')
+                {
+                    stack--;
+                }
+            }
+
+            stop=i;
+
+            for(int ii=start; ii<=stop; ii++)
+            {
+                printf("%c",input[ii]);
+            }
+            printf("\n \n \n \n \n ");
+            i=start;
+        }
+    }
+
+    printf("%d",cnt);
+}
+
+bool psvInainteDeParanteza(char * p, int index);
+
+bool isTestFunction(char * in, int index)
+{
+    char* aux=(char *) malloc(105);
+    strcpy(aux,in);
+
+    int comas=0;
+    bool isFunction=false;
+
+    if(aux[index]=='t' && aux[index+1]=='e' && aux[index+2]=='s' && aux[index+3]=='t') /// scrie test
+    {
+        if(psvInainteDeParanteza(aux,index+4)==true)
+        {
+            /// e variabila
+            return false;
+        }
+        else /// e functie (are paranteze)
+        {
+            // isFunction=true;
+            int ii=index+4;
+
+            while(aux[ii]!=')') /// vad cate virgule exista intre parantezele alea
+            {
+                if(aux[ii]==',')
+                {
+                    comas++;
+                }
+                ii++;
+            }
+            if (comas==0)
+            {
+                printf("0 parameters \n");
+            }
+            else
+            {
+                printf("%d parameters \n",comas+1);
+            }
+            return true;
+        }
+    }
+    return false;
+}
+
+bool psvInainteDeParanteza(char * p, int index)
+{
+    char * pp=(char*) malloc(1005);
+    strcpy(pp,p);
+
+    int i=index;
+
+    while(true)
+    {
+        if(pp[i]==';')
+        {
+            return true;
+        }
+
+        if(pp[i]=='(')
+        {
+            return false;
+        }
+        i++;
+    }
+    return false;
+}
+void querry4(char* in)
+{
+    char* input=(char *) malloc(105);
+    strcpy(input,in);
+
+
+    for(int i=0; input[i]!=NULL; i++)
+    {
+        if(isTestFunction(input,i)==true)
+        {
+            return;
+        }
+    }
+    printf("error, function test does not exist");
+}
+int main()
+{
+    char* input=(char*) malloc(1005);
+    input=readFile(filename);
+
+    querry1(input); // uneori nu merge la comentariile /*  */
+    querry2(file); // merge daca am mai multe file-uri input cu cod, nu m-am ocupat de asta, ideea conteaza
+    querry3(input); /// merge, mai putin daca nu pui acolade; la do{}while() } trebuie sa fie lipita de while()
```

Comment:
Nu stiu daca o sa citesti mesajul asta pt ca nu prea ma pricep pe github, ignora partea a doua a comentariului de la linia 360, pentru ca am da commit-ul si am uitat sa il sterg. Imi merge bine querry 3 si la do while() deci.

Reply:
Și eu ar trebui să ghicesc cerința problemei colegului tău de la tehnică militară din București? :))

Reply:
Ideea care e - eu ți-am spus în comentariile anterioare foarte multe lucruri pe care ar fi trebuit să le schimbi la codul pe care l-ai scris. Nu ai făcut nicio schimbare din câte văd eu. Ce să înțeleg din asta? Nu vreau să îmi pierd timpul dând recomandări de care să nu se țină niciodată cont, cu toate astea, vreau să mă uit peste codul pe care l-ați scris.